pycqed/instrument_drivers/physical_instruments/QuTech_QCC.py

```diff
@@ -0,0 +1,615 @@
+            opcode = opcode.strip().lower()
+
+            # convert the opcode into an integer
+            if opcode.startswith("0x"):
+                base = 16
+            elif opcode.startswith("0o"):
+                base = 8
+            elif opcode.startswith("0b"):
+                base = 2
+            else:
+                base = 10
+            opcode = int(opcode, base)
+
+            if name.startswith("def_q_arg_none"):
+                q_arg[name[16:-2]] = opcode
+            if name.startswith("def_q_arg_tt"):
+                q_arg[name[14:-2]] = opcode
+            if name.startswith("def_q_arg_st"):
+                q_arg[name[14:-2]] = opcode
+
+        print("Instruction      Codewords")
+        for key, value in q_arg.items():
+            print('  {:<10s}:  '.format(key), end = '')
+            self.microcode.print_cs_line_no_header(value)
+            print("")
+
+###############################################################################
+
+#  These are functions which cannot be cast into the standard
+#  form or not that I know of.
+#  They will be added manually using add_parameter explicitly
+
+###############################################################################
+
+    def _upload_instructions(self, filename):
+        """
+        _upload_instructions expects the assembly filename and uses the
+        QISA_Driver as a parser. The QISA_driver then converts it to a binary
+        file which in turn gets read and internally
+        converts the bytes read to a bytearray which is required by
+        binBlockWrite in SCPI.
+        """
+        self.stop()
+        if not isinstance(filename, str):
+            raise ValueError(
+                "The parameter filename type({}) is incorrect. "
+                "It should be str.".format(type(filename)))
+
+        success_parser = self.QISA.assemble(filename)
+
+        if success_parser is not True:
+            print("Error detected while assembling the file {}:".format(
+                filename))
+            print(self.QISA.getLastErrorMessage())
+            raise RuntimeError("Assembling failed.")
+
+        instHex = self.QISA.getInstructionsAsHexStrings(False)
+
+        intarray = []
+        for instr in instHex:
+            intarray.append(int(instr[2:], 16))
+
+        # add a stop instruction at the end of the program
+        intarray.append(0x10000000)
+
+        if len(intarray) > MAX_NUM_INSN:
+            raise OverflowError("Failed to upload instructions: program length ({})"
+                " exceeds allowed maximum value ({}).".format(len(intarray),
+                    MAX_NUM_INSN))
+            return
+
+
+        binBlock = bytearray(array.array('L', intarray))
+        # print("binblock size:", len(binBlock))
+        # write binblock
+        hdr = 'QUTech:UploadInstructions '
+        self.binBlockWrite(binBlock, hdr)
+        # print("QCC: Sending instructions to the hardware finished.")
```

Comment:
Same here, if not needed please remove it

Reply:
Removed